packages/nextalign/include/nextalign/nextalign.h

```diff
@@ -40,6 +40,14 @@ struct Range {
   }
 };
 
+[[nodiscard]] inline Range nucRangeToCodonRange(const Range& range) {
+  return Range{
+    .begin = range.begin / 3,
+    // Make sure the right boundary is aligned to codon boundary
+    .end = (range.end + (3 - range.end % 3) % 3) / 3,
+  };
+}
+
```

Comment:
This is  the new conversion function from nuc to codon positions. note that we adjust `end` to the the next multiple of 3 before the division.